planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp

```diff
@@ -1790,31 +1781,44 @@ bool NormalLaneChange::is_colliding(
   constexpr auto is_safe{true};
   auto current_debug_data = utils::path_safety_checker::createObjectDebug(obj);
   constexpr auto hysteresis_factor{1.0};
-  const auto obj_predicted_paths = utils::path_safety_checker::getPredictedPathFromObj(
-    obj, lane_change_parameters_->safety.collision_check.use_all_predicted_paths);
   const auto safety_check_max_vel = get_max_velocity_for_safety_check();
   const auto & bpp_param = *common_data_ptr_->bpp_param_ptr;
+  const auto th_yaw_diff = common_data_ptr_->lc_param_ptr->safety.collision_check.th_yaw_diff;
+  const auto prepare_duration = lane_change_path.info.duration.prepare;
+
+  auto & debug = current_debug_data.second;
+  debug.ego_predicted_path = ego_predicted_path;
+  debug.current_obj_pose = obj.initial_pose;
+
+  const auto & th_stopped_obj_vel = lane_change_parameters_->safety.th_stopped_object_velocity;
+  const auto & lane_changing_phase_rss_param =
+    (obj.initial_twist.linear.x <= th_stopped_obj_vel)
+      ? lane_change_parameters_->safety.rss_params_for_parked
+      : rss_param;
+
+  for (const auto & obj_path : obj.predicted_paths) {
+    debug.obj_predicted_path = obj_path.path;
+
+    std::vector<autoware_utils_geometry::Polygon2d> collided_polygons{};
+    collided_polygons.reserve(obj_path.path.size());
+    for (const auto & obj_pose_with_poly : obj_path.path) {
+      if (obj_pose_with_poly.time < prepare_duration) {
+        continue;
+      }
 
-  const double velocity_threshold = lane_change_parameters_->safety.th_stopped_object_velocity;
-  const double prepare_duration = lane_change_path.info.duration.prepare;
-  const double start_time = check_prepare_phase ? 0.0 : prepare_duration;
-
-  for (const auto & obj_path : obj_predicted_paths) {
-    utils::path_safety_checker::PredictedPathWithPolygon predicted_obj_path;
-    predicted_obj_path.confidence = obj_path.confidence;
-    std::copy_if(
-      obj_path.path.begin(), obj_path.path.end(), std::back_inserter(predicted_obj_path.path),
-      [&](const auto & entry) {
-        return !(
-          entry.time < start_time ||
-          (entry.time < prepare_duration && entry.velocity < velocity_threshold));
-      });
-
-    const auto collided_polygons = utils::path_safety_checker::get_collided_polygons(
-      lane_change_path.path, ego_predicted_path, obj, predicted_obj_path, bpp_param.vehicle_info,
-      selected_rss_param, hysteresis_factor, safety_check_max_vel,
-      common_data_ptr_->lc_param_ptr->safety.collision_check.th_yaw_diff,
-      current_debug_data.second);
+      CollisionCheckDebug * debug_ptr = collided_polygons.empty() ? &debug : nullptr;
+      const auto & selected_rss_param =
+        obj_pose_with_poly.time < prepare_duration
```

Comment:
CPPCheck found that this condition cannot be true since we `continue` at the beginning of the loop if it is true.

Reply:
done in https://github.com/autowarefoundation/autoware_universe/pull/11307/commits/5c0170f5792327b41abfb0824e31e6a817544c0f